routers/common/middleware.go

```diff
@@ -81,3 +83,31 @@ func Middlewares() []func(http.Handler) http.Handler {
 	})
 	return handlers
 }
+
+func stripSlashesMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		var urlPath string
+		rctx := chi.RouteContext(req.Context())
+		if rctx != nil && rctx.RoutePath != "" {
+			urlPath = rctx.RoutePath
+		} else {
+			urlPath = req.URL.Path
+		}
+
+		var sanitizedPath string
+		if len(urlPath) > 1 {
+			sanitizedPath = path.Clean(urlPath)
```

Comment:
[`path.Clean`](https://pkg.go.dev/path#Clean) is the wrong function for this job. It has other side-effects involving with how file systems works, which Gitea "shouldn't" care about. 
This increased restrictiveness is causing the tests to fails.

Reply:
@Gusted Sure yeah. Removed it's usage.

Reply:
@Gusted The tests still fail.

Reply:
I see, never ran tests on non-ascii characters. Code will need adjustments for that likely, you can add some debug in here and run the test via: `make test-sqlite#TestNonasciiBranches` to see what's going on.